core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -812,10 +812,10 @@ private[spark] object MapOutputTracker extends Logging {
         logError(errorMessage)
         throw new MetadataFetchFailedException(shuffleId, startPartition, errorMessage)
       } else {
-        for (part <- startPartition until endPartition) {
-          splitsByAddress.getOrElseUpdate(status.location, ArrayBuffer()) +=
-            ((ShuffleBlockId(shuffleId, mapId, part), status.getSizeForBlock(part)))
-        }
+        val totalSize: Long = (startPartition until endPartition).map(status.getSizeForBlock).sum
+        splitsByAddress.getOrElseUpdate(status.location, ArrayBuffer()) +=
+          ((ShuffleBlockId(shuffleId, mapId, startPartition, endPartition - startPartition),
+            totalSize))
```

Comment:
This is going to create some very heavy shuffle fetches - and looks incorrect.
This merge should not be happening here, but in `ShuffleBlockFetcherIterator`